File: files/Portfolio/alpen-aerial-project/alpen-aerial-project.js
$(document).ready(function(){
	var addCount = 0;
	<?php 
	    $companies = scandir('../alpenaerial/');
	    $companies = json_encode($companies);
	    echo "var companies = ". $companies . ";\n";
	?>

    clients = {};
    
	for(var i = 0; i < companies.length; i++){
	    if(companies[i] != '.' && companies[i] != '..'){
	        client = companies[i].replace('.txt','').replaceAll('_',' ').split('-')[0];
	        project = companies[i].replace('.txt','').replaceAll('_',' ').split('-')[1];
		    $('#clientSelect').append('<option>'+client+'<\option>');
		    clients[client] == null ? clients[client] = [project] : clients[client].push(project);
		    if($('#clientSelect').children().last().val() == ''){
		        $('#clientSelect').children().last().remove();
		    }
	    }
	}
	
	$('#fill').hide();
	$('.projectSelect').hide();
	
	<?php echo "var txt_url = '". $txt_url . "';\n"; ?>
	<?php echo "var company_name = '". $company_name . "';\n"; ?>	
	<?php echo "var project_name = '". $project_name . "';\n"; ?>

    if(txt_url){
        $('#fill').show();
	    $('.projectSelect').show();
        <?php $content = file_get_contents($txt_url); ?>
    	<?php echo "var formContent = '". $content . "';\n"; ?>
    	$('#clientSelect').val(company_name);
		for(var i = 0; i < clients[$('#clientSelect').val()].length; i++){
			company = $('#clientSelect').val();
			project = clients[company][i];
			$('#projectSelect').append('<option>'+project+'<\option>');
			if($('#clientSelect').children().last().val() == ''){
	            $('#clientSelect').children().last().remove();
	        }
		}
		$('#projectSelect').val(project_name);
		var content = formContent.split('xxiixx');
		var content_0 = content[0].split('xxixx');

		$('select[name="name"]').val(content_0[0]);
		$('input[name="project"]').val(content_0[1]);
		$('input[name="company"]').val(content_0[2]);
		$('input[name="contact"]').val(content_0[3]);
		$('input[name="clientEmail"]').val(content_0[4]);
		$('input[name="clientPhone"]').val(content_0[5]);
		$('textarea[name="description"]').val(content_0[6]);
		var addCount = 0;
		for(var i = 1; i < content.length - 1; i++){
		    content_num = content[i].split('xxixx');
		    $('#formTable_1').before('<tr id="formTableAdd_'+addCount+'"><td id="<td id="formTableAdd_'+addCount+'_0"><p class="button buttonRedRing del">X</p></td><td id="formTableAdd_'+addCount+'_1"><input type="text" name="old_'+addCount+'_1" required></td><td id="formTableAdd_'+addCount+'_2"><select type="text" name="old_'+addCount+'_2"><option>Text</option><option>Image</option><option>Text + Location</option><option>Text + Image</option></select></td><td id="formTableAdd_'+addCount+'_3"><select type="text" name="old_'+addCount+'_3" required><option>Clickable</option><option>Not Clickable</option></select></td><td id="formTableAdd_'+addCount+'_4"><input type="text" name="old_'+addCount+'_4"></td><td id="formTableAdd_'+addCount+'_5"><input type="text" name="old_'+addCount+'_5"></td><td id="formTableAdd_'+addCount+'_6"><input type="text" name="old_'+addCount+'_6"></td><td id="formTableAdd_'+addCount+'_7"><textarea type="text" name="old_'+addCount+'_7"></textarea></td><td id="formTableAdd_'+addCount+'_8"><textarea type="text" name="old_'+addCount+'_8"></textarea></td></tr>');
		    addCount++;
		    $('input[name="old_'+(i-1).toString()+'_1').val(content_num[0]);
		    $('select[name="old_'+(i-1).toString()+'_2').val(content_num[1]);
		    $('select[name="old_'+(i-1).toString()+'_3').val(content_num[2]);
		    $('input[name="old_'+(i-1).toString()+'_4').val(content_num[3]);
		    $('input[name="old_'+(i-1).toString()+'_5').val(content_num[4]);
		    $('input[name="old_'+(i-1).toString()+'_6').val(content_num[5]);
		    $('textarea[name="old_'+(i-1).toString()+'_7').val(content_num[6]);
		    $('textarea[name="old_'+(i-1).toString()+'_8').val(content_num[7]);
		}
    }

	function projectf(company, project){
		file = '../alpenaerial/'+company.replaceAll(' ', '_')+'-'+project.replaceAll(' ', '_')+'.txt';
		$('input[name="txt_url"]').val(file);
		$('input[name="company_name"]').val(company);
		$('input[name="project_name"]').val(project);
		document.getElementById('hiddenForm').submit();
	}
	var company = '';
	var project = '';

	$('#clientSelect').bind('input propertychange', function() {
		if($(this).val() == ''){
			$('#fill').hide();
			$('.projectSelect').hide();
		}
		else if($(this).val() == 'New Client'){
			$('#fill').show();
			$('input[text]').val('');
			$('textarea').val('');
			$('.projectSelect').hide();
		}
		else{
			$('#fill').hide();
			$('.projectSelect').show();
			for(var i = 0; i < clients[$('#clientSelect').val()].length; i++){
				company = $('#clientSelect').val();
				project = clients[company][i];
				$('#projectSelect').append('<option>'+project+'<\option>');
				if($('#projectSelect').children().last().val() == ''){
		            $('#projectSelect').children().last().remove();
		        }
			}
		}
	});

	$('#projectSelect').bind('input propertychange', function() {
		if($(this).val() == ''){
			$('#fill').hide();
		}
		else if($(this).val() == 'New Project'){
			$('#fill').show();
			$('input[type=text]').val('');
			$('textarea').val('');
		}
		else{
		    $('#fill').show();
		    projectf(company, project);
		}
	});


	$('#addButton').click(function(){
		$('#formTable_1').before('<tr id="formTableAdd_'+addCount+'"><td id="<td id="formTableAdd_'+addCount+'_0"><p class="button buttonRedRing del">X</p></td><td id="formTableAdd_'+addCount+'_1"><input type="text" name="in_'+addCount+'_1" required></td><td id="formTableAdd_'+addCount+'_2"><select type="text" name="in_'+addCount+'_2"><option>Text</option><option>Image</option><option>Text + Location</option><option>Text + Image</option></select></td><td id="formTableAdd_'+addCount+'_3"><select type="text" name="in_'+addCount+'_3" required><option>Clickable</option><option>Not Clickable</option></select></td><td id="formTableAdd_'+addCount+'_4"><input type="text" name="in_'+addCount+'_4"></td><td id="formTableAdd_'+addCount+'_5"><input type="text" name="in_'+addCount+'_5"></td><td id="formTableAdd_'+addCount+'_6"><input type="text" name="in_'+addCount+'_6"></td><td id="formTableAdd_'+addCount+'_7"><textarea type="text" name="in_'+addCount+'_7"></textarea></td><td id="formTableAdd_'+addCount+'_8"><textarea type="text" name="in_'+addCount+'_8"></textarea></td></tr>');
		addCount++;
	});
	
	$("#formTable").on("click", ".del", function() {
		$(this).parents('tr').remove();
	});
});

File: files/Portfolio/alpen-aerial-project/alpen-aerial-project.py
import modules as mx
import sys
import os
homedir = os.getcwd() + r'/files/Portfolio/'
sys.path.append(homedir)
data = mx.Data()
body = mx.Body()

from header import *
from alpenaerialproject import *
from forms import *
from visitInfo import *

data.title = "Dre Barrera | Alpen Aerial Project Brief"
data.author = "Andres Barrera"

### OBJECTS ###
# Main
main = mx.C()
container = mx.C()
captionT = mx.T()
formC = mx.C()
title = mx.T()
contact = mx.C()
form = mx.X()
formTable = mx.Table()
th0 = mx.X()
th1 = mx.T()
th2 = mx.T()
th3 = mx.T()
th4 = mx.T()
th5 = mx.T()
th6 = mx.T()
th7 = mx.T()
th8 = mx.T()
addButton = mx.T()
formend = mx.X()

### CONTENT ###
# Body
body.content = [main]

# Main
main.content = [nav, container]

# Container
container.content = [title, captionT, formC]
captionT.content = 'This form is for submitting project details to Alpen Aerial&#39;s design team. If you are submitting a revision to previously submitted project information, please include "REVISED" in the Project Name. If any media is necessary for the project, please email it to <a href="mailto:drebarrera@yahoo.com" class="pseudoLink">drebarrera@yahoo.com</a> <em>before or immediately after</em> the request is sent. If necessary, give us a call at <a href="tel:12105856279" class="pseudoLink">+1 (210) 585-6279</a> during business hours. Don&#39;t be shy to leave a voicemail! We will do our best to get back to you within 24 hours on the next business day.<br><br>'
formC.content = [contact]
title.content = 'Alpen Aerial Project Information Form'
contact.content = [form, formTable, formend]
form.content = '<form method="post" action="index.php"><label for="name">Submitter</label><select name="name" required><option>Client</option><option>Steve Kohl</option><option>Asher Noël</option><option>Andres Barrera</option></select><label for="project">Project Name</label><input type="text" name="project" required><label for="company">Client Company</label><input type="text" name="company"><label for="contact">Client Contact Name</label><input type="text" name="contact"><label for="clientEmail">Contact Email</label><input type="text" name="clientEmail"><label for="clientPhone">Contact Phone</label><input type="text" name="clientPhone"><label for="description">Project Description</label><textarea name="description" type="text" placeholder="Please provide a full description of the project. This includes where to find necessary media and project background information. Any unclear information will be resolved with the client contact." style="height:200px" required></textarea><!--table id="table"><tr><th>Icon</th><th>Type</th><th>Clickable/Not Clickable</th><th>Latitude/Longitude (optional)</th><th>Display Information (if clickable)</th><th>Other Descriptional Information (Media URLS,Positional Info,etc.)</th></tr></table><input type="submit" value="Submit" name="submit"><p style="color: white;"><?=$thankyou ?></p></form-->'

# FormTable
formTable.content = [[th0,th1,th2,th3,th4,th5,th6,th7,th8],[th0,addButton]]
th1.content = 'Icon Name'
th2.content = 'Icon Type'
th3.content = 'Clickable / Not Clickable'
th4.content = 'Icon Text Content (optional)'
th5.content = 'Icon Image Description / URL (optional)'
th6.content = 'Latitude/Longitude (optional)'
th7.content = 'Display Information (if clickable)'
th8.content = 'Other Descriptional Information (Media URLS,Positional Info,etc.)'
addButton.content = 'Add New Icon +'
formend.content = '<input type="submit" value="Submit" name="submit"><p style="color: white;"><?=$thankyou ?></p></form>'

### PROPERTIES ###
# Body
body.background_color = "#fefefe"

# Main
main.id = "main"
main.background_color = ""

# Caption
container.id = "container"
container.background_color = ""

# Form
formC.id = "form"
formC.background_color = ""
captionT.id = "caption"
title.id = "title"
contact.id = "contact"
contact.background_color = "#005580"

#FormTable
formTable.background_color = ""
formTable.id = "formTable"
addButton.cl = 'button buttonRed'
addButton.id = 'addButton'
